plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -245,10 +245,12 @@ func (s *SQLServer) Gather(acc telegraf.Accumulator) error {
 }
 
 func (s *SQLServer) gatherServer(server string, query Query, acc telegraf.Accumulator) error {
-	// deferred connection opening
+	// connection object
 	var conn *sql.DB
 
-	if strings.Contains(strings.ToLower(server), strings.ToLower("password=")) {
+	// setup connection based on authentication
+	rx := regexp.MustCompile("(?i)\\b(?:;(Password=\\w+))\\b")
```

Comment:
Not sure but is `dsn="Password=verysecret"` also a valid DSN (without the semicolon)?

Reply:
I've updated regex condition used for AAD auth applicability where I look for Authentication parameter instead of Password. However with regards to your concern about well-formed dsn or connection string, the intention of this check if the connection string provides a way to identify auth mechanism and not if its well-formed. We could check that but it should be part of validating in the input config file instead of this check alone.

Just for clarity, the new regex I'm using now "(?i)(^|[;])(?:(\s*Authentication=Active Directory Integrated\s*))([;]|$)" would take care of scenarios where the connection string contains the "Authentication=Active Directory Integrated" string anywhere (at the start or at the end or in the middle somewhere).